src/snowcli/cli/nativeapp/run_processor.py

```diff
@@ -197,9 +206,176 @@ def _create_dev_app(self, diff: DiffResult) -> None:
+        """
+        Get an existing version, if present, by the same name for an application package.
+        It executes a 'show versions like ... in application package' query and returns the result as single row, if one exists.
+        """
+        with self.use_role(self.package_role):
+            show_obj_query = f"show versions like '{unquote_identifier(version)}' in application package {self.package_name}"
+            show_obj_cursor = self._execute_query(
+                show_obj_query, cursor_class=DictCursor
+            )
+
+            if show_obj_cursor.rowcount is None:
+                raise SnowflakeSQLExecutionError(show_obj_query)
+
+            show_obj_row = find_first_row(
+                show_obj_cursor,
+                lambda row: row[VERSION_COL] == unquote_identifier(version),
+            )
+
+            return show_obj_row
+
+    def upgrade_app(
+        self,
+        policy: PolicyBase,
+        is_interactive: bool,
+        version: Optional[str] = None,
+        patch: Optional[str] = None,
+    ):
+
+        patch_clause = f"patch {patch}" if patch else ""
+        using_clause = f"using version {version} {patch_clause}" if version else ""
+
+        with self.use_role(self.app_role):
+
+            # 1. Need to use a warehouse to create an application instance
+            try:
+                if self.application_warehouse:
+                    self._execute_query(f"use warehouse {self.application_warehouse}")
+            except ProgrammingError as err:
+                generic_sql_error_handler(
+                    err=err, role=self.app_role, warehouse=self.application_warehouse
+                )
+
+            # 2. Check for an existing application by the same name
+            show_app_row = self.get_existing_app_info()
+
+            # 3. If existing application is found, perform a few validations and upgrade the instance.
+            if show_app_row:
+
+                drop_app = False
+                # We skip comment check here, because prod apps/pre-existing apps may not be created by the CLI.
+                # Check for the right owner
+                ensure_correct_owner(
+                    row=show_app_row, role=self.app_role, obj_name=self.app_name
+                )
+
+                # If all the above checks are in order, proceed to upgrade
+                try:
+                    self._execute_query(
+                        f"alter application {self.app_name} upgrade {using_clause}"
+                    )
+                    return
+
+                except ProgrammingError as err:
+                    # The existing app was created from a different process.
+                    if err.errno in UPGRADE_RESTRICTION_CODES:
+                        print(err.msg)
+                        user_prompt = "Do you want the CLI to drop the existing application and recreate it?"
+                        if not policy.should_proceed(user_prompt):
```

Comment:
This prompt asks the user if they are okay dropping an existing application in case it fails to upgrade. Such a situation can occur if:
1. Existing app is in loose files, and you want to upgrade to versioned install or release directives install.
2. Existing app is in versioned dev mode, and you want to upgrade to the other two.
3. Existing app is through release directive install, and you want to upgrade to the other two.
If the user agrees, the existing app is dropped and a new app with the same name is installed according to the flags passed in by the user.

Reply:
--force check is then embedded in it, I suppose? - if that's the case, feel free to resolve.